src/test/java/org/opensearch/knn/index/LuceneEngineIT.java

```diff
@@ -499,13 +499,22 @@ private void baseQueryTest(SpaceType spaceType) throws Exception {
         }
 
         validateQueries(spaceType, FIELD_NAME);
+        validateQueries(spaceType, FIELD_NAME, Map.of("ef_search", 100));
     }
 
     private void validateQueries(SpaceType spaceType, String fieldName) throws Exception {
+        validateQueries(spaceType, fieldName, null);
+    }
+
+    private void validateQueries(SpaceType spaceType, String fieldName, Map<String, ?> methodParameters) throws Exception {
 
         int k = LuceneEngineIT.TEST_INDEX_VECTORS.length;
         for (float[] queryVector : TEST_QUERY_VECTORS) {
-            Response response = searchKNNIndex(INDEX_NAME, new KNNQueryBuilder(fieldName, queryVector, k), k);
+            Response response = searchKNNIndex(
+                INDEX_NAME,
+                KNNQueryBuilder.builder().fieldName(fieldName).k(k).vector(queryVector).methodParameters(methodParameters).build(),
```

Comment:
For the end to end test, can we use XContentBuilder to build the query?

https://github.com/opensearch-project/k-NN/blob/main/src/testFixtures/java/org/opensearch/knn/KNNRestTestCase.java#L213

Reply:
Its much more convenient to use KNNBuilder. It converts to XContent [underneath](https://github.com/opensearch-project/k-NN/blob/main/src/testFixtures/java/org/opensearch/knn/KNNRestTestCase.java#L194). 

Not opposed to it but any specific reason we should move to xcontent for happy case tests?

Reply:
Using XContentBuilder can better mock end-user API calls by simulating JSON content creation and ensuring end-to-end functionality.

KNNBuilder is an internal method, and relying on it in integration test has the possibility to lead to fragile tests that may break with internal changes, even if the public API remains stable.

Reply:
Fair, will change it